juju/juju
# Fix unmanaged "controller" namespace clash with "controller" model.

While bootstrapping the namespace of the controller is unknown also during creating of the controller model's broker the namespace is unknown.

Previously the name of the controller model was used ("controller"). But the controller namespace is in the form of controller-<controller-name> (e.g. controller-microk8s-localhost). This means the target k8s cluster could legitimately have a namespace called "controller" that is not managed by juju.

This fixes this clash by using an empty string namespace for controller models when bootstrapping and forcing namespace resolution to always use the slow path for controller broker instances after bootstrap.

## QA steps

Should not see the error `namespace "controller" may already be in use`

```
$ juju bootstrap microk8s
$ juju deploy istio-pilot --trust
# wait for Istio-pilot to be ready
$ juju config istio-pilot trust
true
$ juju destroy-controller microk8s
```

## Documentation changes

N/A

## Bug reference

N/A
